parquet-column/src/main/java/org/apache/parquet/column/values/bytestreamsplit/ByteStreamSplitValuesReaderForDouble.java

```diff
@@ -33,19 +32,7 @@ public ByteStreamSplitValuesReaderForDouble() {
 
   @Override
   public double readDouble() {
-    try {
-      gatherElementDataFromStreams(valueByteBuffer);
-      long value = (long)(valueByteBuffer[0] & 0xFF) |
-                   ((long)(valueByteBuffer[1] & 0xFF) << 8) |
-                   ((long)(valueByteBuffer[2] & 0xFF) << 16) |
-                   ((long)(valueByteBuffer[3] & 0xFF) << 24) |
-                   ((long)(valueByteBuffer[4] & 0xFF) << 32) |
-                   ((long)(valueByteBuffer[5] & 0xFF) << 40) |
-                   ((long)(valueByteBuffer[6] & 0xFF) << 48) |
-                   ((long)(valueByteBuffer[7] & 0xFF) << 56);
-      return Double.longBitsToDouble(value);
-    } catch (IOException | ArrayIndexOutOfBoundsException ex) {
-      throw new ParquetDecodingException("Could not read double.", ex);
-    }
+    gatherElementDataFromStreams(valueByteBuffer);
+    return ByteBuffer.wrap(valueByteBuffer).order(ByteOrder.LITTLE_ENDIAN).getDouble();
```

Comment:
I don't agree with @shangxinli. The `ByteBuffer` functionality might be readable but here performance is critical. You can speed up this line by having the `ByteBuffer` object created at the class level and set the order so you only invoke `getDouble()` here but even in this case a lower level implementation will perform better.
To reuse code, I would suggest to take a look on [ByteUtils](https://github.com/apache/parquet-mr/blob/master/parquet-common/src/main/java/org/apache/parquet/bytes/BytesUtils.java).

Reply:
I see, using ByteUtils is likely better. I am not aware what the jit compiler will generate and I have not measured the performance impact of the proposed change. But using those static functions to serialize/deserialize the values is probably better.
I have my own benchmark I will use in the near future to check whether performance is bad.
Will address this in the evening.